src/app/public/modules/modal/modal-scroll-shadow.directive.ts

```diff
@@ -0,0 +1,148 @@
+import {
+  Directive,
+  ElementRef,
+  EventEmitter,
+  HostListener,
+  OnDestroy,
+  OnInit,
+  Output
+} from '@angular/core';
+
+import {
+  Subject
+} from 'rxjs';
+
+import {
+  takeUntil
+} from 'rxjs/operators';
+
+import {
+  MutationObserverService
+} from '@skyux/core';
+
+import {
+  SkyTheme,
+  SkyThemeService
+} from '@skyux/theme';
+
+import {
+  SkyModalScrollShadowEventArgs
+} from './modal-scroll-shadow-event-args';
+
+/**
+ * @internal
+ */
+@Directive({
+  selector: '[skyModalScrollShadow]'
+})
+export class SkyModalScrollShadowDirective implements OnInit, OnDestroy {
+
+  @Output()
+  public skyModalScrollShadow = new EventEmitter<SkyModalScrollShadowEventArgs>();
```

Comment:
Should we be completing `EventEmitter`s on destroy? I've seen us follow two different patterns here, so we're kind of inconsistent.

Reply:
I don't think it's necessary because all subscribers to the `EventEmitter` will be unsubscribed when the host component is destroyed. I've never seen documentation or examples where components or directives explicitly complete their `EventEmitter`s on destroy.

Reply:
I'm also not finding examples where `EventEmitter`s are destroyed. Ok, thanks for your thoughts. Lets move on.